python/lsst/ap/association/diaPipe.py

```diff
@@ -109,6 +109,43 @@ def __init__(self, *, config=None):
         if not config.doWriteAssociatedSources:
             self.outputs.remove("associatedDiaSources")
 
+    def adjustQuantum(self, datasetRefMap: pipeBase.InputQuantizedConnection):
+        """Override to make adjustments to `lsst.daf.butler.DatasetRef` objects
+        in the `lsst.daf.butler.core.Quantum` during the graph generation stage
+        of the activator.
+
+        This implementation checks to make sure that the filters in the dataset
+        are compatible with AP processing as set by the Apdb/DPDD schema.
+
+        Parameters
+        ----------
+        datasetRefMap : `NamedKeyDict`
+            Mapping from dataset type to a `set` of
+            `lsst.daf.butler.DatasetRef` objects
+
+        Returns
+        -------
+        datasetRefMap : `NamedKeyDict`
+            Mapping of input with assurances that bands incompatible with the
+            Apdb are present.
+
+        Raises
+        ------
+        ValueError
+            Raises if a data ref in the quantum has a band not available in the
+            Apdb.
+        """
+        refs = datasetRefMap[self.diffIm.name]
+        for ref in refs:
+            if ref.dataId["band"] not in self.config.validBands:
+                raise ValueError(
+                    f"Requested '{ref.dataId['band']}' not in "
+                    "DiaPipelineConfig.validBands. To process bands not in the "
+                    "standard Rubin set (ugrizy) you must add the band to the "
+                    "validBands list in DiaPipelineConfig and add the "
+                    "appropriate columns to the Apdb schema.")
```

Comment:
I think this ValueError should include the dataId and/or ref that failed. Otherwise the user might not know exactly where that particular band came from.

Reply:
I don't think that would be too helpful. As is it would just report the first dataId it encountered which would possibly just prompt the user to remove them one after the other vs removing them all.

Reply:
I'll add a --show-data to the error message.

Reply:
Actually, re-reading my error message. I'm fine with keeping it as is.